diff --git a/env/fs_posix.cc b/env/fs_posix.cc
index 82fb9fba3..fe9b52dbb 100644
--- a/env/fs_posix.cc
+++ b/env/fs_posix.cc
@@ -94,6 +94,9 @@ struct LockHoldingInfo {
 static std::map<std::string, LockHoldingInfo> locked_files;
 static port::Mutex mutex_locked_files;
 
+#if defined(IOS_CROSS_COMPILE)
+static int LockOrUnlock(int fd, bool lock) { return 0; }
+#else
 static int LockOrUnlock(int fd, bool lock) {
   errno = 0;
   struct flock f;
@@ -106,6 +109,7 @@ static int LockOrUnlock(int fd, bool lock) {
 
   return value;
 }
+#endif
 
 class PosixFileLock : public FileLock {
  public:
